rq/registry.py

```diff
@@ -334,6 +333,66 @@ def cleanup(self):
         pass
 
 
+class QueuedJobRegistry(BaseRegistry):
+    key_template = 'rq:queued:{0}'
+
+    def get_expired_job_ids(self, timestamp=None):
+        raise NotImplementedError
+
+    def cleanup(self):
+        """This method is only here to prevent errors because this method is
+        automatically called by `count()` and `get_job_ids()` methods
+        implemented in BaseRegistry."""
+        pass
+
+    def requeue_stuck_jobs(self):
+        """This method requeues jobs that are popped off the queue but still belong to QueuedJobRegistry
+        It is not defined in cleanup since we don't want this being called everytime count or get_job_ids is called
+        """
+        job_ids = self.get_job_ids()
+        for job_id in job_ids:
+            # If job was enqueued AFTER the front of the queue it must have already been dequeued
+            # This is faster than seeing if the job is in the queue directly.
+            front_timestamp = self._get_front_queue_timestamp() 
```

Comment:
We can move this out of the loop so we don't keep getting new timestamp for every single job processed.